script/tool/lib/src/version_check_command.dart

```diff
@@ -205,16 +156,85 @@ class VersionCheckCommand extends PluginCommand {
                 NextVersionType.BREAKING_MAJOR) {
           final String error = '$pubspecPath breaking change detected.\n'
               'Breaking changes to platform interfaces are strongly discouraged.\n';
-          final Colorize redError = Colorize(error)..red();
-          print(redError);
-          throw ToolExit(1);
+          PrintErrorAndExit(errorMessage: error);
         }
       } on io.ProcessException {
         print('Unable to find pubspec in master for $pubspecPath.'
             ' Safe to ignore if the project is new.');
       }
     }
 
+    await for (Directory plugin in getPlugins()) {
+      await _checkVersionsMatch(plugin);
+    }
+
     print('No version check errors found!');
   }
+
+  Future<void> _checkVersionsMatch(Directory plugin) async {
+    // get version from pubspec
+    final String packageName = plugin.basename;
+    print('-----------------------------------------');
+    print(
+        'Checking the first version listed in CHANGELOG.MD matches the version in pubspec.yaml for $packageName.');
+
+    final Pubspec pubspec = _tryParsePubspec(plugin);
+    if (pubspec == null) {
+      final String error = 'Cannot parse version from pubspec.yaml';
+      PrintErrorAndExit(errorMessage: error);
+    }
+    final Version fromPubspec = pubspec.version;
+
+    // get first version from CHANGELOG
+    final File changelog = plugin.childFile('CHANGELOG.md');
+    final List<String> lines = changelog.readAsLinesSync();
+    String firstLineWithText;
+    final Iterator iterator = lines.iterator;
+    while (iterator.moveNext()) {
+      final String currentStriptEmptySpaces =
+          (iterator.current as String).replaceAll(' ', '');
+      if (currentStriptEmptySpaces != null &&
+          currentStriptEmptySpaces.isNotEmpty) {
+        firstLineWithText = iterator.current;
+        break;
+      }
+    }
+    // Remove all leading mark down syntax from the version line.
+    final String versionString = firstLineWithText.split(' ').last;
+    Version fromChangeLog = Version.parse(versionString);
+    if (fromChangeLog == null) {
+      final String error =
+          'Cannot find version on the first line of CHANGELOG.md';
```

Comment:
This path isn't printing what it found and tried to parse (related to previous review comment).

Reply:
done